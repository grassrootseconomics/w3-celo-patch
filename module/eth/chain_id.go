package eth

import (
	"github.com/grassrootseconomics/w3-celo-patch/internal/module"
	"github.com/grassrootseconomics/w3-celo-patch/w3types"
)

// ChainID requests the chains ID.
func ChainID() w3types.CallerFactory[uint64] {
	return module.NewFactory(
		"eth_chainId",
		nil,
		module.WithRetWrapper(module.HexUint64RetWrapper),
	)
}
