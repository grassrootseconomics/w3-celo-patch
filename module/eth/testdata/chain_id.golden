> {"jsonrpc":"2.0","id":1,"method":"eth_chainId"}
< {"jsonrpc":"2.0","id":1,"result":"0x1"}
